src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -20,227 +129,2379 @@ const ActorInit En_Ossan_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5594.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5680.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC576C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5858.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5900.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5A28.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5C24.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5C9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5D40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5DAC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC5EF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC60E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC6148.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC61B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC6208.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC6258.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC62C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC62F4.s")
-
+#endif
+
+u8 func_80AC3D18(EnOssan* this, u8 arg1) {
+    u8 i;
+
+    if (this->unk_252 & 1) {
+        for (i = arg1 + 1; i < arg1 + 4; i += 2) {
+            if (this->unk_200[i] != NULL) {
+                return i;
+            }
+        }
+        for (i = arg1; i < arg1 + 4; i += 2) {
+            if (this->unk_200[i] != NULL) {
+                return i;
+            }
+        }
+    } else {
+        for (i = arg1; i < arg1 + 4; i += 2) {
+            if (this->unk_200[i] != NULL) {
+                return i;
+            }
+        }
+        for (i = arg1 + 1; i < arg1 + 4; i += 2) {
+            if (this->unk_200[i] != NULL) {
+                return i;
+            }
+        }
+    }
+    return -1;
+}
+
+u8 func_80AC3E50(EnOssan* this, u8 arg1, u8 arg2) {
+    u8 tmp = arg2 + 4;
+
+    while ((arg1 >= arg2) && (arg1 < tmp)) {
+        arg1 -= 2;
+        if (((arg1 >= arg2) && (arg1 < tmp)) && (this->unk_200[arg1] != NULL)) {
+            return arg1;
+        }
+    }
+    return -1;
+}
+
+u8 func_80AC3ED8(EnOssan* this, u8 arg1, u8 arg2) {
+    while (arg1 < arg2) {
+        arg1 += 2;
+        if ((arg1 < arg2) && (this->unk_200[arg1] != NULL)) {
+            return arg1;
+        }
+    }
+    return -1;
+}
+
+void func_80AC3F38(EnOssan* this, GlobalContext* globalCtx) {
+    if (gSaveContext.rupees < D_80AC88EC[this->unk_1EB]) {
+        // What?!
```

Comment:
these look like mask prices. id name the array and change the numbers to decimal in the array
and remove the text below